src/all/hennojin/src/eu/kanade/tachiyomi/extension/all/hennojin/Hennojin.kt

```diff
@@ -66,46 +74,47 @@ class Hennojin(override val lang: String, suffix: String) : ParsedHttpSource() {
     override fun searchMangaFromElement(element: Element) =
         popularMangaFromElement(element)
 
-    override fun mangaDetailsRequest(manga: SManga) =
-        GET("https://hennojin.com" + manga.url, headers)
-
     override fun mangaDetailsParse(document: Document) =
         SManga.create().apply {
-            description = document.selectFirst(
+            description = document.select(
                 ".manga-subtitle + p + p",
-            )?.html()?.replace("<br> ", "\n")
+            ).joinToString("\n") {
+                it.run {
+                    select(Evaluator.Tag("br")).prepend("\\n")
+                    this.text().replace("\\n", "\n").replace("\n ", "\n")
+                }
+            }.trim()
             genre = document.select(
                 ".tags-list a[href*=/parody/]," +
                     ".tags-list a[href*=/tags/]," +
                     ".tags-list a[href*=/character/]",
-            )?.joinToString { it.text() }
-            artist = document.select(
+            ).joinToString { it.text() }
+            artist = document.selectFirst(
                 ".tags-list a[href*=/artist/]",
-            )?.joinToString { it.text() }
-            author = document.select(
+            )?.text()
+            author = document.selectFirst(
                 ".tags-list a[href*=/group/]",
-            )?.joinToString { it.text() } ?: artist
+            )?.text() ?: artist
             status = SManga.COMPLETED
         }
 
-    override fun fetchChapterList(manga: SManga) =
-        Request.Builder().url(manga.thumbnail_url!!)
-            .head().build().run(client::newCall)
-            .asObservableSuccess().map { res ->
-                SChapter.create().apply {
-                    name = "Chapter"
-                    url = manga.reader
-                    date_upload = res.date
-                    chapter_number = -1f
-                }.let(::listOf)
-            }!!
-
-    override fun pageListRequest(chapter: SChapter) =
-        GET("https://hennojin.com" + chapter.url, headers)
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup(response.body.string())
+        val date = Request.Builder().url(document.selectFirst(".manga-thumbnail > img")!!.attr("src"))
+            .head().build().run(client::newCall).execute().date
+        return document.select("a:contains(Read Online)").map {
+            SChapter.create().apply {
+                url = it.attr("href").replace("&view=page", "&view=multi")
```

Comment:
Use `HttpUrl`? Something like:
```kt
setUrlWithoutDomain(it
    ?.absUrl("href")
    ?.toHttpUrlOrNull()
    ?.newBuilder()
    ?.removeAllQueryParameters("view")
    ?.addQueryParameter("view", "multi")
    ?.build()
    ?.toString()
    ?: it.absUrl("href")
)
```

Reply:
For reference, this worked for me:
```kt
override fun chapterListParse(response: Response): List<SChapter> {
    val document = response.asJsoup(response.body.string())
    val date = Request.Builder().url(document.selectFirst(".manga-thumbnail > img")!!.attr("src"))
        .head().build().run(client::newCall).execute().date
    return document.select("a:contains(Read Online)").map {
        SChapter.create().apply {
            setUrlWithoutDomain(
                it
                    ?.absUrl("href")
                    ?.toHttpUrlOrNull()
                    ?.newBuilder()
                    ?.removeAllQueryParameters("view")
                    ?.addQueryParameter("view", "multi")
                    ?.build()
                    ?.toString()
                    ?: it.absUrl("href"),
            )
            name = "Chapter"
            date_upload = date
            chapter_number = -1f
        }
    }
}
```


<details>

<summary>This is however incorrect, as it would overwrite whatever `setUrlWithoutDomain` sets `url` to:</summary>

```kt
override fun chapterListParse(response: Response): List<SChapter> {
    val document = response.asJsoup(response.body.string())
    val date = Request.Builder().url(document.selectFirst(".manga-thumbnail > img")!!.attr("src"))
        .head().build().run(client::newCall).execute().date
    return document.select("a:contains(Read Online)").map {
        SChapter.create().apply {
            url = setUrlWithoutDomain(
                it
                    ?.absUrl("href")
                    ?.toHttpUrlOrNull()
                    ?.newBuilder()
                    ?.removeAllQueryParameters("view")
                    ?.addQueryParameter("view", "multi")
                    ?.build()
                    ?.toString()
                    ?: it.absUrl("href"),
            ).toString()
            name = "Chapter"
            date_upload = date
            chapter_number = -1f
        }
    }
}
```
</details>

Reply:
> For reference, this worked for me:
> 
> ```kotlin
> override fun chapterListParse(response: Response): List<SChapter> {
>     val document = response.asJsoup(response.body.string())
>     val date = Request.Builder().url(document.selectFirst(".manga-thumbnail > img")!!.attr("src"))
>         .head().build().run(client::newCall).execute().date
>     return document.select("a:contains(Read Online)").map {
>         SChapter.create().apply {
>             setUrlWithoutDomain(
>                 it
>                     ?.absUrl("href")
>                     ?.toHttpUrlOrNull()
>                     ?.newBuilder()
>                     ?.removeAllQueryParameters("view")
>                     ?.addQueryParameter("view", "multi")
>                     ?.build()
>                     ?.toString()
>                     ?: it.absUrl("href"),
>             )
>             name = "Chapter"
>             date_upload = date
>             chapter_number = -1f
>         }
>     }
> }
> ```
> 
> This is however incorrect, as it would overwrite whatever `setUrlWithoutDomain` sets `url` to:
> ```kotlin
> override fun chapterListParse(response: Response): List<SChapter> {
>     val document = response.asJsoup(response.body.string())
>     val date = Request.Builder().url(document.selectFirst(".manga-thumbnail > img")!!.attr("src"))
>         .head().build().run(client::newCall).execute().date
>     return document.select("a:contains(Read Online)").map {
>         SChapter.create().apply {
>             url = setUrlWithoutDomain(
>                 it
>                     ?.absUrl("href")
>                     ?.toHttpUrlOrNull()
>                     ?.newBuilder()
>                     ?.removeAllQueryParameters("view")
>                     ?.addQueryParameter("view", "multi")
>                     ?.build()
>                     ?.toString()
>                     ?: it.absUrl("href"),
>             ).toString()
>             name = "Chapter"
>             date_upload = date
>             chapter_number = -1f
>         }
>     }
> }
> ```

sr i forgot to clean up my cache mihon app ＞﹏＜